modules/apps/headless/headless-admin-content/headless-admin-content-impl/src/main/java/com/liferay/headless/admin/content/internal/dto/v1_0/util/StructuredContentVersionInformationBuilder.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package com.liferay.headless.admin.content.internal.dto.v1_0.util;
+
+import com.liferay.headless.admin.content.dto.v1_0.StructuredContentVersionInformation;
+import com.liferay.headless.admin.content.dto.v1_0.VersionInformation;
+import com.liferay.headless.delivery.dto.v1_0.StructuredContent;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+/**
+ * @author Luis Miguel Barcos
+ */
+public class StructuredContentVersionInformationBuilder {
+
+
+	public StructuredContentVersionInformationBuilder(
+		StructuredContent structuredContent) {
+
+		_structuredContent = structuredContent;
+		_structuredContentVersionInformation =
+			new StructuredContentVersionInformation();
+	}
+
+	public StructuredContentVersionInformation build() {
+		_buildStructuredContentVersionInformationParent(
+			_structuredContent, _structuredContentVersionInformation);
+
+		return _structuredContentVersionInformation;
+	}
+
+	public StructuredContentVersionInformationBuilder setVersionInformation(
+		VersionInformation versionInformation) {
+
+		_structuredContentVersionInformation.setVersionInformation(
+			versionInformation);
+
+		return this;
+	}
+
+	private void _buildStructuredContentVersionInformationParent(
```

Comment:
I don't understand the name here... Parent? Maybe with a from is more clear? or maybe removing parent at all

Reply:
`_buildStructuredContentVersionInformationFromParent` is not exactly what the method does because it is not "building" a new object from another, is refilling the parent of StructuredContentVersionInformation

So, I've tried to point that in the name, but it's still confusing.

Reply:
Maybe you can just remove the Parent name